pkg/controller/account/account_controller.go

```diff
@@ -631,6 +625,81 @@ func (r *ReconcileAccount) statusUpdate(reqLogger logr.Logger, account *awsv1alp
 	return err
 }
 
+func (r *ReconcileAccount) setStateFailed(reqLogger logr.Logger, currentAcctInstance *awsv1alpha1.Account, msg string) error {
+	reqLogger.Info(msg)
+	// Upodate the account status (state and condition) to Failed
+	utils.SetAccountStatus(reqLogger, currentAcctInstance, msg, v1alpha1.AccountFailed, AccountFailed)
+	err := r.statusUpdate(reqLogger, currentAcctInstance)
+	if err != nil {
+		return err
+	}
+
+	// Set a failure condition in the accountClaim
+	err = r.setAccountClaimError(reqLogger, currentAcctInstance, msg)
+	if err != nil {
+		return err
+	}
```

Comment:
I don't think we should be doing this here at all.

To the extent possible, $foo_controller ought to be exclusively responsible for updating $foo CRs; otherwise it's hard to understand which one is supposed to "win".

And in this case, when you update the Account's Status, it'll trigger a reconcile in the accountclaim_controller, which will see the Account's updated status and make the same status update anyway.

[Later] Okay, I see this is latent, account_controller was doing `accountClaimBYOCError` before. So this might not be the PR to make that change.

Reply:
I agreed with your point - I am loathe to have one controller make changes to another, but this was the suggested way forward by the team for resolving this.  I've got some notes for future improvements that I'd like make cards for, and this is one of them.